### PR TITLE
[luci/pass] Introduce CanonicalizePass with Pad

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce CanonicalizePass with Pad.paddings canonicalize.

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #13490
- from #13492

File: compiler/luci/pass/src/CanonicalizePass.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  // TODO relocate to helpers/CreateCircleConst.h when necessary
+  auto num_elements = paddings->size<loco::DataType::S64>();
+  for (uint32_t i = 0; i < num_elements; i++)
+  {
+    auto v64 = paddings->at<loco::DataType::S64>(i);
+    auto hval = static_cast<int64_t>(std::numeric_limits<int32_t>::max());
+    auto lval = static_cast<int64_t>(std::numeric_limits<int32_t>::lowest());
+    CHECK_OR_FALSE(v64 < hval);
+    CHECK_OR_FALSE(v64 > lval);
```
Comment (jinevening) [inline compiler/luci/pass/src/CanonicalizePass.cpp:51]:
```suggestion
    CHECK_OR_FALSE(v64 <= hval);
    CHECK_OR_FALSE(v64 >= lval);
```

File: compiler/luci/pass/include/luci/Pass/CanonicalizePass.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#include <logo/Pass.h>
+
+namespace luci
+{
+
+/**
+ * @brief  Class to canoncalize CircleNodes
+ *
+ */
+struct CanonicalizePass final : public logo::Pass
```
Comment (jinevening) [inline compiler/luci/pass/include/luci/Pass/CanonicalizePass.h:29]: This pass name seems too general. This name more fits a stage, i.e., a sequence of passes.

How about renaming to CanonicalizePaddingDtypePass, or simply ConvertS64PaddingToS32Pass ?

File: compiler/luci/pass/src/CanonicalizePass.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  bool changed = false;
+  for (auto node : loco::active_nodes(loco::output_nodes(g)))
+  {
+    if (auto pad = dynamic_cast<luci::CirclePad *>(node))
+    {
+      if (paddings_to_s32(pad))
+        changed = true;
+    }
+
+    // TODO add more canonicalization
```
Comment (jinevening) [inline compiler/luci/pass/src/CanonicalizePass.cpp:95]: Putting all canonicalization tasks in a single pass seems to make this pass to have too many responsibilities.

File: compiler/luci/pass/include/luci/Pass/CanonicalizePass.h (hunk 2 above)
Comment (seanshpark) [reply compiler/luci/pass/include/luci/Pass/CanonicalizePass.h:29]: Named this pass to do some multiple things as canonicalization.
This pass will run always without option.

File: compiler/luci/pass/src/CanonicalizePass.cpp (hunk 3 above)
Comment (seanshpark) [reply compiler/luci/pass/src/CanonicalizePass.cpp:95]: Yes, its' about canonicalization.
When there are too many things to do, someone can split to some categorize.

**Review by jinevening — APPROVED** [review]
https://github.com/Samsung/ONE/pull/13503/files#r1690674624 violates single responsibility principle, which would leave a technical debt.

You can go as-is if you wish.

**Pull request merged**